### PR TITLE
Fixes(#2877)

### PR DESCRIPTION
**Pull request by siddhant2002**
### **Describe your change:**

- [x] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by siddhant2002** [comment]
@siriak please check the updated pr for maximum subarray sum

**Review by siriak — CHANGES_REQUESTED** [review]
Thanks for contributing, this algorithm is already present here https://github.com/TheAlgorithms/Java/blob/9fb3364cccc5373c32dbdb661f773c5ee753c77a/src/main/java/com/thealgorithms/dynamicprogramming/KadaneAlgorithm.java as far as I can tell, but the old implementation is missing tests, so your improvements are welcome. Could you consolidate your and existing implementations into one?

**Comment by 794547955** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by CrpMihasha** [comment]
亲，邮件已收到

File: src/test/java/com/thealgorithms/others/KadaneAlogrithmTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.thealgorithms.dynamicprogramming.KadaneAlgorithm;
+public class KadaneAlogrithmTest {
+    @Test
+	void testForOneElement() 
+	{
+        int a[]={-1};
+		assertTrue(KadaneAlgorithm.max_Sum(a));
```
[the remark this answers is not in the document]
Comment (siddhant2002) [reply src/test/java/com/thealgorithms/others/KadaneAlogrithmTest.java:11]: Basically it's checking with the predicted answer

**Pull request closed without merging**